api/routes/index.js

```diff
@@ -14,4 +14,69 @@ router.get('/', async (req, res) => {
   }
 });
 
+//retrieving ALL co-ops in our coops table
+router.get('/coops', async (req, res) => {
+  try {
+    const query = await db.query(`SELECT * FROM coops;`);
+    res.send(query.rows); 
+  } catch (error) {
+    console.log(error.stack);
+  }
+});
+
+//retrieve the co-ops and their starred attribute
+router.get('/getStarred/:starredId', async (req, res) => {
+  const {starredId} = req.params
+
+  console.log("PARAMS = " + starredId)
+  try {
+    //DONE: use get the id's of the starred co-ops
+    // `SELECT *, starred_coop_id
+    // FROM coops LEFT JOIN stars 
+    // on coops.id = $1;`,
+    const query = await db.query(
+      `SELECT starred_coop_id 
+      FROM stars 
+      WHERE starrer_coop_id = $1;`,
+      [starredId]
+    );
+    console.log(query.rows);
+    res.send(query.rows);
+  } catch (error) {
+    console.log(error.stack);
+  }
+});
+
+//post new starred
+router.post('/addStar', async (request, response) => {
+  try {
+    const query = await db.query(
+      `INSERT INTO stars (starred_coop_id, starrer_coop_id))
+        VALUES ($1, $2);`,
+      [request.body.starred_coop_id, request.body.starrer_coop_id]
+    );
+    response.send(query.rows);
+  } catch (error) {
+    console.log(error.stack);
+  }
+});
+
+//delete star
+router.delete('/delete', async (req, res) => {
+  const { starredId } = req.params;
+  const { starrerId } = req.params;
+  try {
+    const query = await db.query(
+      //DONE: delete row from stars table
+      `DELETE FROM stars 
+      WHERE starred_coop_id = $1 
+      AND starrer_coop_id = $2;`,
+      [request.body.starredId, request.body.starrerId]
```

Comment:
`request` doesn't exist in this context (check the parameters of this callback) -- you should be able to use `starrerId` and `starredId` that you declared above